src/main/java/com/example/b2b_opportunities/Service/PositionService.java

```diff
@@ -256,4 +265,20 @@ private void deactivateProjectIfNoActivePositions(Project project) {
             projectRepository.save(project);
         }
     }
+
+    private void setPositionDefaultStatus(Position position) {
+        position.setIsActive(true);
+
+        position.setStatus(positionStatusRepository.findByName("Opened"));
+    }
+
+    public void editPositionStatus(Long id, PositionStatus positionStatus, Authentication authentication) {
+        validateUserAndCompany(authentication);
+        Position position = getPositionOrThrow(id);
```

Comment:
We should check if the position is in a project that belongs to the same company that the user is in. User X should not be able to edit the position status of company that belongs to User Y